UnityAds/src/main/java/com/mopub/mobileads/UnityInterstitial.java

```diff
@@ -152,101 +150,85 @@ protected void show() {
         }
     }
 
-    @Override
-    protected void onInvalidate() {
-        UnityAds.removeListener(UnityInterstitial.this);
-    }
-
-    @Nullable
-    @Override
-    protected LifecycleListener getLifecycleListener() {
-        return null;
-    }
-
-    @NonNull
-    @Override
-    protected String getAdNetworkId() {
-        return mPlacementId != null ? mPlacementId : "";
-    }
-
-    @Override
-    protected boolean checkAndInitializeSdk(@NonNull final Activity activity, @NonNull final AdData adData) {
-        return false;
-    }
-
-    @Override
-    public void onUnityAdsReady(String placementId) {
-        // no-op
-    }
-
-    @Override
-    public void onUnityAdsStart(String placementId) {
-        MoPubLog.log(SHOW_SUCCESS, ADAPTER_NAME);
-
-        if (mInteractionListener != null) {
-            mInteractionListener.onAdShown();
-            mInteractionListener.onAdImpression();
-        }
-    }
-
-    @Override
-    public void onUnityAdsFinish(String placementId, UnityAds.FinishState finishState) {
-        if (mInteractionListener != null) {
-            if (finishState == UnityAds.FinishState.ERROR) {
+    /**
+     * IUnityAdsShowListener instance. Contains logic for callbacks when showing ads.
+     */
+    private IUnityAdsShowListener mUnityShowListener = new IUnityAdsShowListener() {
+        @Override
+        public void onUnityAdsShowFailure(String placementId, UnityAds.UnityAdsShowError error, String message) {
+            if (error == UnityAdsShowError.VIDEO_PLAYER_ERROR) {
                 MoPubLog.log(CUSTOM, ADAPTER_NAME, "Unity interstitial video encountered a playback error for " +
                         "placement " + placementId);
                 MoPubLog.log(SHOW_FAILED, ADAPTER_NAME,
                         MoPubErrorCode.VIDEO_PLAYBACK_ERROR.getIntCode(),
                         MoPubErrorCode.VIDEO_PLAYBACK_ERROR);
-
-                mInteractionListener.onAdFailed(MoPubErrorCode.VIDEO_PLAYBACK_ERROR);
+                if (mInteractionListener != null) {
+                    mInteractionListener.onAdFailed(MoPubErrorCode.VIDEO_PLAYBACK_ERROR);
+                }
             } else {
-                MoPubLog.log(CUSTOM, ADAPTER_NAME, "Unity interstitial video completed for placement " + placementId);
-                mInteractionListener.onAdDismissed();
+                if (mContext != null) {
+                    // Lets Unity Ads know when ads fail to show
+                    MediationMetaData metadata = new MediationMetaData(mContext);
+                    metadata.setMissedImpressionOrdinal(++missedImpressionOrdinal);
+                    metadata.commit();
+                }
+
+                MoPubLog.log(CUSTOM, ADAPTER_NAME, "Unity rewarded video failed to show with error message: " + message);
+                MoPubLog.log(SHOW_FAILED, ADAPTER_NAME,
+                        MoPubErrorCode.UNSPECIFIED.getIntCode(),
```

Comment:
We probably want to make the error code more accurate, rather than just `UNSPECIFIED `